src/io.c

```diff
@@ -2174,7 +2174,7 @@ static StructInitInfo module = {
     // init struct using C99 designated initializers; for a full list of
     // fields, please refer to the definition of StructInitInfo
     .type = MODULE_BUILTIN,
-    .name = "scanner",
+    .name = "io",
```

Comment:
Just so I understand this better:
The documentation for `StructInitInfo` under `.name` says: 

> name of the module: filename with ".c" or library filename    

There is both a `scanner.c` and `io.c`, so that's fine. Clearly we want to refer to `io` here, I understand that. But why doesn't this trigger problems if the name corresponds to the wrong `.c` file?

Reply:
Because we don't have code that verifies that no module name is used twice.

But there are functions which lookup modules by name, and if you used those with the name `"scanner"`, they would return whichever module comes first in the list of modules; and that actually is `io`, not `scanner`. Luckily, this doesn't matter at all, because those functions are not actually called that way (and they might never be). So this had no real effect, for now. But it might one day...

For now, the only "visible" effect is when using debugging tools, like the `-D` option, which currently gives this output during startup:
```$ ./gap -D
#I  InitInfo(builtin gvars)
#I  InitInfo(builtin objects)
#I  InitInfo(builtin profile)
#I  InitInfo(builtin hookintrprtr)
#I  InitInfo(builtin tracing)
#I  InitInfo(builtin scanner)
#I  InitInfo(builtin scanner)
#I  InitInfo(builtin read)
#I  InitInfo(builtin calls)
...
```
Note how `scanner` is listed twice.

This is also how I noticed this.

Reply:
Thank you!